publishAgents.py

```diff
@@ -0,0 +1,74 @@
+import json
+import requests
+import time
+import csv
+from datetime import datetime
+
+secretsVersion = input('To edit production server, enter the name of the \
+secrets file: ')
+if secretsVersion != '':
+    try:
+        secrets = __import__(secretsVersion)
+        print('Editing Production')
+    except ImportError:
+        secrets = __import__('secrets')
+        print('Editing Development')
+else:
+    secrets = __import__('secrets')
+    print('Editing Development')
+
+startTime = time.time()
+
+baseURL = secrets.baseURL
+user = secrets.user
+password = secrets.password
+repository = secrets.repository
+
+auth = requests.post(baseURL + '/users/' + user + '/login?password='
+                     + password).json()
+
+session = auth['session']
+headers = {'X-ArchivesSpace-Session': session,
+           'Content_Type': 'application/json'}
+
+agentTypes = ['people', 'families', 'corporate_entities', 'software']
+
+date = datetime.now().strftime('%Y-%m-%d %H.%M.%S')
+f = csv.writer(open('publishedAgents' + date + '.csv', 'w'))
+f.writerow(['uri'] + ['post'])
+f2 = csv.writer(open('errorsAgents' + date + '.csv', 'w'))
+f2.writerow(['uri'] + ['post'])
+
+for agentType in agentTypes:
+    endpoint = '/agents/' + agentType + '?all_ids=true'
+    ids = requests.get(baseURL + endpoint, headers=headers).json()
+    counter = 0
+    for id in ids:
+        counter += 1
+        print(counter)
+        id = str(id)
+        print(baseURL + '/agents/' + agentType + '/' + id)
+        output = requests.get(baseURL + '/agents/' + agentType + '/' + id,
+                              headers=headers)
+
+        output = output.json()
+        output['publish'] = True
+        asRecord = json.dumps(output)
+        post = requests.post(baseURL + '/agents/' + agentType + '/' + id,
+                             headers=headers, data=asRecord)
+        print(post.status_code)
+        if post.status_code != 200:
```

Comment:
I'm not sure how the CSV files are being used, but I think I'd probably suggest something like the following instead of checking that the status code is 200:
```python
try:
    post.raise_for_status()
except requests.HTTPError as e:
    f2.writerow([id] + [e])
    print('error')
```
This may give you more information about the error than just using the string representation of the response object, which is pretty bare bones.

Reply:
Great suggestion, I need to explore more robust error-reporting. Thanks!